main.go

```diff
@@ -102,308 +124,285 @@ func exportEnvironmentWithEnvman(keyStr, valueStr string) error {
-			if err := retry.Times(retryCount).Wait(waitTime).Try(func(attempt uint) error {
-				if attempt > 0 {
-					log.Warnf("Retrying...")
-				}
+	log.Printf("=> %s\n   value: %s\n", env, l)
+	if err := exportEnvironmentWithEnvman(env, l); err != nil {
+		return fmt.Errorf("envman export failed, error: %v", err)
+	}
+	return nil
+}
 
-				submoduleErr := git.SubmoduleUpdate()
-				if submoduleErr != nil {
-					log.Warnf("Attempt %d failed:", attempt+1)
-					fmt.Println(submoduleErr.Error())
-				}
+func main() {
+	originPresent, err := isOriginPresent(configs.CloneIntoDir, configs.RepositoryURL)
+	if err != nil {
+		fail("Can't check if origin is presented, error: %v", err)
+	}
 
-				return submoduleErr
-			}); err != nil {
-				log.Errorf("Failed, error: %s", err)
-				os.Exit(1)
-			}
+	if originPresent && configs.ResetRepository == "yes" {
+		if err := resetRepo(); err != nil {
+			fail("Can't reset repository, error: %v", err)
 		}
+	}
 
-		log.Infof("Exporting git logs")
+	// Create directory if not exist
+	if err := os.MkdirAll(configs.CloneIntoDir, 0755); err != nil {
+		fail("Can't create directory (%s), error: %v", configs.CloneIntoDir, err)
+	}
 
-		if commitHash, err := git.LogCommitHash(); err != nil {
-			log.Errorf("Git log failed, error: %s", err)
-			os.Exit(1)
-		} else {
-			log.Printf("=> GIT_CLONE_COMMIT_HASH")
-			log.Printf("   value: %s", commitHash)
-			fmt.Println()
+	if err := run(Git.Init()); err != nil {
+		fail("Can't init repository, error: %v", err)
+	}
 
-			if err := exportEnvironmentWithEnvman("GIT_CLONE_COMMIT_HASH", commitHash); err != nil {
-				log.Warnf("envman export failed, error: %s", err)
-			}
+	if !originPresent {
+		if err := run(Git.RemoteAdd("origin", configs.RepositoryURL)); err != nil {
+			fail("Can't add remote repository (%s), error: %v", configs.RepositoryURL, err)
 		}
+	}
 
-		if commitMessageSubject, err := git.LogCommitMessageSubject(); err != nil {
-			log.Errorf("Git log failed, error: %s", err)
-			os.Exit(1)
-		} else {
-			log.Printf("=> GIT_CLONE_COMMIT_MESSAGE_SUBJECT")
-			log.Printf("   value: %s", commitMessageSubject)
-			fmt.Println()
-
-			if err := exportEnvironmentWithEnvman("GIT_CLONE_COMMIT_MESSAGE_SUBJECT", commitMessageSubject); err != nil {
-				log.Warnf("envman export failed, error: %s", err)
-			}
+	if err := runWithRetry(func() *command.Model {
+		if configs.CloneDepth != "" {
+			return Git.Fetch("--depth=" + configs.CloneDepth)
 		}
+		return Git.Fetch()
+	}); err != nil {
+		fail("Fetch failed, error: %v", err)
+	}
 
-		if commitMessageBody, err := git.LogCommitMessageBody(); err != nil {
-			log.Errorf("Git log failed, error: %s", err)
-			os.Exit(1)
-		} else {
-			log.Printf("=> GIT_CLONE_COMMIT_MESSAGE_BODY")
-			log.Printf("   value: %s", commitMessageBody)
-			fmt.Println()
-
-			if err := exportEnvironmentWithEnvman("GIT_CLONE_COMMIT_MESSAGE_BODY", commitMessageBody); err != nil {
-				log.Warnf("envman export failed, error: %s", err)
+	if isPR() {
```

Comment:
the 3 case if pr:

- configs.ManualMerge == "no" && configs.PullRequestMergeBranch != ""
- configs.ManualMerge == "no" && hasDiffFile
- others (nit github pr nor bitbucket -> configs.ManualMerge does not matter)